tsl/src/compression/compression.c

```diff
@@ -297,14 +321,44 @@ compress_chunk(Oid in_table, Oid out_table, const ColumnCompressionInfo **column
 
 	TupleDesc in_desc = RelationGetDescr(in_rel);
 	TupleDesc out_desc = RelationGetDescr(out_rel);
+	in_rel_index_iods = RelationGetIndexList(in_rel);
+	foreach (lc, in_rel_index_iods)
+	{
+		Oid index_oid = lfirst_oid(lc);
+		Relation index_rel = index_open(index_oid, AccessShareLock);
+		if (index_rel->rd_att->natts >= n_keys)
+		{
+			for (i = 0; i < n_keys; i++)
+			{
+				bool compression_col_is_orderby = COMPRESSIONCOL_IS_ORDER_BY(keys[i]);
+				bool compression_col_is_segmentby = COMPRESSIONCOL_IS_SEGMENT_BY(keys[i]);
+				if (namestrcmp((Name) &keys[i]->attname,
+							   NameStr(index_rel->rd_att->attrs[i].attname)) == 0 &&
+					(compression_col_is_orderby || compression_col_is_segmentby))
+				{
+					if (compression_col_is_segmentby)
+						indexscan_direction = BackwardScanDirection;
+					else
+						indexscan_direction = ForwardScanDirection;
```

Comment:
Why should `indexscan_direction` be specified by is_segmentby or is_orderby? ISTM, if we want to have same output as tuplesort (ascending AFAIK), then we should inspect the `matched_index_rel` and check if it's an asc/desc index and set the scan direction to FORWARD if it's asc or to BACKWARD if it's descending when we do index_getnext_slot on it.